backtracking/graph_coloring.cpp

```diff
@@ -1,18 +1,36 @@
+/**
+ * @file
+ * @brief brief one line description here
+ */
 #include <iostream>
 
-/*
- * Number of vertices in the graph
+// created a namespace to ensure all globals and functions are
+// restricted to this file
+namespace {
+/** A utility function to print solution
+ * @param color description
+ * @param V number of vertices in the graph
  */
-#define V 4
+void printSolution(const int *color, int V) {
+    std::cout << "Following are the assigned colors\n";
+    for (int i = 0; i < V; i++) {
+        std::cout << color[i];
+    }
+    std::cout << "\n";
+}
 
-/**
- * Prototypes
+/** A utility function to check if the current color assignment is safe for
+ * vertex v
+ * @tparam V number of vertices in the graph
+ * @param v description
+ * @param graph description
+ * @param color description
+ * @param c description
+ * @returns `true` if ....
+ * @returns `false` if ....
  */
-void printSolution(int color[]);
-
-/* A utility function to check if the current color assignment
-   is safe for vertex v */
-bool isSafe(int v, bool graph[V][V], int color[], int c) {
+template <int V>
+bool isSafe(int v, const bool graph[V][V], const int *color, int c) {
```

Comment:
Will LGTM consider this as a short variable name?

Reply:
The variable name is local to the function and not a global variable. If you READ the LGTM reports, they are regarding *global* variables